addons/survey/static/src/js/survey_form.js

```diff
@@ -1109,6 +1112,15 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
                 $(dependingQuestion).find('textarea').val('');
             }
         }
+        // Sections visibility
+        if (this.options.questionsLayout !== 'one_page'){
+            return;
+        }
```

Comment:
I would instead put the following code inside that (inversed) if.

A return in the middle of a method is a bit misleading, as people could add code after your section and wonder why it's not executed.
(It's ok for global pre-conditions, as done here above, though).